aptos-move/framework/src/built_package.rs

```diff
@@ -265,6 +266,12 @@ impl BuiltPackage {
             }
         }
 
+        if let Some(model_options) = model.get_extension::<Options>() {
+            if model_options.experiment_on(Experiment::STOP_AFTER_EXTENDED_CHECKS) {
+                std::process::exit(0)
+            }
+        }
```

Comment:
Is this supposed to stop the process entirely?  Is this what's expected?

What if I'm publishing a contract, and I enable this, so it just stops and says it succeeds, though no upload occurs?

Reply:
This "experiment" is an experiment/debugging flag, except perhaps for some different compilation cases.

I was hoping to quietly avoid generating bytecode files when `aptos move lint` is run, but there's no way to have "extended checks" without writing some files.  The metadata is "injected" after extended checks, so perhaps users are already trained that the existence of bytecode files doesn't mean a successful compilation.  I think the solution will be to just treat `aptos move lint` as `aptos move build --lint` for now.

Reply:
Yeah, it appears that is the way how model building and extended checks work, which is also intentionally for the Prover, as that one is supposed to work on generated code.

Reply:
Ok, removed the `STOP_AFTER_EXTENDED_CHECKS` flag since it doesn't help that much anyway.

Reply:
No, I'm going to keep `STOP_AFTER_EXTENDED_CHECKS` for the linter, as we're turning of optimizations when running it and hopefully the lack of metadata will prevent users from accidentally uploading it.

Reply:
Actually the `deploy` command anyway runs compilation first, so I think there is no danger of accidentally uploading.